react/index.tsx

```diff
@@ -6,54 +6,78 @@ const CSS_HANDLES = ['container'] as const
 
 enum Positions {
   BOTTOM = 'bottom',
+  TOP = 'top',
 }
 
 interface Props {
   position?: Positions
+  verticalSpacing?: number
 }
 
 interface StorefrontComponent extends FC<Props> {
   schema?: any
 }
 
-const StickyLayoutComponent: StorefrontComponent = ({
-  children,
-  position,
-}) => {
-  const container = useRef<HTMLDivElement>(null)
-  const handles = useCssHandles(CSS_HANDLES)
-
-  if (position !== Positions.BOTTOM) {
-    // Only 'bottom' position supported for now!
+const StickyLayoutComponent: StorefrontComponent = ({ children, position, verticalSpacing = 0 }) => {
+  if (!position || (position !== Positions.BOTTOM && position !== Positions.TOP)) {
     return <Fragment>{children}</Fragment>
   }
 
-  const handleScroll = useCallback((e?: Event) => {
-    const target = (e && e.target) as HTMLDivElement | null
-    if (!container.current) {
+  const containerReference = useRef<HTMLDivElement>(null)
+  const container = useRef<HTMLDivElement>(null)
+  const handles = useCssHandles(CSS_HANDLES)
+  const parsedVerticalSpacing = verticalSpacing && verticalSpacing
+
+  const handleScroll = useCallback(() => {
```

Comment:
This function isn't debounced, shouldn't it? It will be called on every scroll position, that might be really intense.

Reply:
Done.